jacoco-maven-plugin/src/org/jacoco/maven/ReportAggregateMojo.java

```diff
@@ -0,0 +1,163 @@
+ * From those projects class and source files as well as JaCoCo execution data
+ * files will be collected. This also allows to create coverage reports when
+ * tests are in separate projects than the code under test, for example in case
+ * of integration tests.
+ * </p>
+ * 
+ * <p>
+ * Using the dependency scope allows to distinguish projects which contribute
+ * execution data but should not be part of the report itself:
+ * </p>
+ * 
+ * <ul>
+ * <li><code>compile</code>: Project source and execution data is included in
+ * the report.</li>
+ * <li><code>test</code>: Only execution data is considered for the report.</li>
+ * </ul>
+ * 
+ * @goal report-aggregate
+ * @since 0.7.7
+ */
+public class ReportAggregateMojo extends ReportMojo {
+
+	/**
+	 * The projects in the reactor.
+	 * 
+	 * @parameter expression="${reactorProjects}"
+	 * @readonly
+	 */
+	private List<MavenProject> reactorProjects;
+
+	@Override
+	public boolean canGenerateReport() {
+		if (skip) {
+			getLog().info("Skipping JaCoCo execution");
+			return false;
+		}
+		return true;
+	}
+
+	@Override
+	protected void executeReport(final Locale locale)
+			throws MavenReportException {
+		loadExecutionData();
+		try {
+			final IReportVisitor visitor = createVisitor(locale);
+			visitor.visitInfo(sessionInfoStore.getInfos(),
+					executionDataStore.getContents());
+			createReport(visitor);
+			visitor.visitEnd();
+		} catch (final IOException e) {
+			throw new MavenReportException("Error while creating report: "
+					+ e.getMessage(), e);
+		}
+	}
+
+	@Override
+	void createReport(final IReportGroupVisitor visitor) throws IOException {
+		final IReportGroupVisitor group = visitor.visitGroup(getProject()
+				.getName());
+		for (final MavenProject dependency : findDependencies(Artifact.SCOPE_COMPILE)) {
+			createReportForProject(dependency, group);
+		}
+	}
+
+	void createReportForProject(final MavenProject project,
+			final IReportGroupVisitor visitor) throws IOException {
+
+		final FileFilter fileFilter = new FileFilter(this.getIncludes(),
+				this.getExcludes());
+		final BundleCreator creator = new BundleCreator(project, fileFilter,
+				getLog());
+		final IBundleCoverage bundle = creator.createBundle(executionDataStore);
+		// TODO use source encoding of target project
+		final SourceFileCollection locator = new SourceFileCollection(
+				getCompileSourceRoots(project), sourceEncoding);
+		checkForMissingDebugInformation(bundle);
+		visitor.visitBundle(bundle, locator);
+	}
+
+	@Override
+	void loadExecutionData() throws MavenReportException {
+		final ExecFileLoader loader = new ExecFileLoader();
+		for (final MavenProject dependency : findDependencies(
+				Artifact.SCOPE_COMPILE, Artifact.SCOPE_TEST)) {
+			// TODO Use configured location from project
+			// TODO Clarify when to include exec data from integration tests
+			final File execFile = new File(
+					dependency.getBuild().getDirectory(), "jacoco.exec");
```

Comment:
As a last comment concerning goals for a JaCoCo user it could be more convenient to write
```
<executions>
<execution>
<goals>
<goal>report-aggregate</goal>
<goal>report-aggregate-integration</goal>
</goals>
</execution>
</executions>
```
instead of fiddling around with multiple executions. And we already have `report-integration`, so it would be more consistent.